torch/legacy/nn/PixelShuffle.py

```diff
@@ -0,0 +1,135 @@
+        self.upscaleFactorSquared = self.upscaleFactor * self.upscaleFactor
+        self._intermediateShape = None
+        self._outShape = None
+        self._shuffleOut = None
+
+    def updateOutput(self, input):
+
+        if hasattr(self, '_intermediateShape') is False:
+            self._intermediateShape = None
+        if hasattr(self, '_outShape') is False:
+            self._outShape = None
+        if hasattr(self, '_shuffleOut') is False:
+            self._shuffleOut = None
+
+        if self._intermediateShape is None:
+            self._intermediateShape = torch.Tensor(6, 0)
+        if self._outShape is None:
+            self._outShape = torch.Tensor()
+        if self._shuffleOut is None:
+            self._shuffleOut = input.new()
+
+        batched = False
+        batchSize = 1
+        inputStartIdx = 0
+        outShapeIdx = 0
+
+        if len(input.size()) == 4:
+            batched = True
+            batchSize = input.size(0)
+            inputStartIdx = 1
+            outShapeIdx = 1
+            self._outShape = torch.Tensor([0, 0, 0, 0])
+            self._outShape[0] = batchSize
+
+        else:
+            self._outShape = torch.Tensor([0, 0, 0])
+
+        channels = input.size(inputStartIdx) / self.upscaleFactorSquared
+        inHeight = input.size(inputStartIdx + 1)
+        inWidth = input.size(inputStartIdx + 2)
+
+        (self._intermediateShape)[0] = batchSize
+        (self._intermediateShape)[1] = channels
+        (self._intermediateShape)[2] = self.upscaleFactor
+        (self._intermediateShape)[3] = self.upscaleFactor
+        (self._intermediateShape)[4] = inHeight
+        (self._intermediateShape)[5] = inWidth
+
+        self._outShape[outShapeIdx] = channels
+        self._outShape[outShapeIdx + 1] = inHeight * self.upscaleFactor
+        self._outShape[outShapeIdx + 2] = inWidth * self.upscaleFactor
+
+        tmp = None
+        tmp = self._intermediateShape.type(torch.IntTensor)
+        tmp = torch.Size((tmp[0], tmp[1], tmp[2], tmp[3], tmp[4], tmp[5]))
+        inputView = input.view(tmp)
+
+        self._shuffleOut.resize_(inputView.size(0), inputView.size(1),
+                                 inputView.size(4), inputView.size(2),
+                                 inputView.size(5), inputView.size(3))
+        self._shuffleOut.copy_(inputView.permute(0, 1, 4, 2, 5, 3))
+
+        tmp = None
+        if len(input.size()) == 4:
+            tmp = self._outShape.type(torch.IntTensor)
+            tmp = torch.Size((tmp[0], tmp[1], tmp[2], tmp[3]))
+        else:
+            tmp = self._outShape.type(torch.IntTensor)
+            tmp = torch.Size((tmp[0], tmp[1], tmp[2]))
+        self.output = self._shuffleOut.view(tmp)
+
+        print("the r factor is : ", self.upscaleFactor)
+        print("the input is: ", input.size())
+        print("input type: ", input.type())
+        print("the output is: ", self.output.size())
+        print("output type: ", self.output.type())
```

Comment:
You need to remove prints used for debugging

Reply:
Done